test/e2e/index.ts

```diff
@@ -1441,6 +1441,52 @@ test(
   })
 );
 
+test(
+  "client reconnects websocket if closed before making a zome call",
+  withConductor(ADMIN_PORT, async (t) => {
+    const { cell_id, client, admin } = await installAppAndDna(ADMIN_PORT, true);
+    await admin.authorizeSigningCredentials(cell_id);
+    await client.client.close();
+    const call = client.callZome({
+      cell_id,
+      zome_name: TEST_ZOME_NAME,
+      fn_name: "bar",
+      provenance: cell_id[1],
+      payload: null,
+    });
+    try {
+      await call;
+      t.pass("websocket was reconnected successfully");
+    } catch (error) {
+      t.fail("websocket was not reconnected");
+    }
+  })
+);
+
+test(
+  "client fails to reconnect to websocket if closed before making a zome call and does not end up in a loop",
+  withConductor(ADMIN_PORT, async (t) => {
+    const { cell_id, client, admin } = await installAppAndDna(ADMIN_PORT);
+    await admin.authorizeSigningCredentials(cell_id);
+    await client.client.close();
+    const call = client.callZome({
+      cell_id,
+      zome_name: TEST_ZOME_NAME,
+      fn_name: "bar",
+      provenance: cell_id[1],
+      payload: null,
+    });
+    try {
+      await call;
+      t.fail(
+        "reconnecting to websocket should have failed due to an invalid token."
+      );
+    } catch (error) {
```

Comment:
assert the failure reason here?

Reply:
As written above it throws the "client disconnected with pending requests" error and it seemed to me like it doesn't make sense to match on that.

Reply:
Let's figure out the conversation above first. But I think it's always worth asserting the error message to make sure it doesn't become some array access out of bounds error. We need to know that the test is doing what it claims to do

Reply:
Okay, I will assert that it's the client closed with pending requests then.

Reply:
6dcaeec4305cd76be0200ff9657f28189063b1e8